src/intermediates/add_data.jl

```diff
@@ -1,26 +1,29 @@
 function add_data!(v::VegaVisualization;
-	                          name::AbstractString = string("table_", lowercase(randstring(5))),
+                              name::AbstractString = string("table_", lowercase(randstring(5))),
                               x::AbstractVector = Int[],
-	                          y::AbstractVector = Int[],
+                              x2::AbstractVector = Int[],
+                              y::AbstractVector = Int[],
                               y2::AbstractVector = Int[],
-	                          group::AbstractVector = Int[])
+                              group::AbstractVector = Int[])
 
     #Check arrays all of same length
     n = length(x)
 
     if n != length(y) || (n != length(group) && length(group) != 0)
-    	error("x, y and group must all have the same length")
+        error("x, y and group must all have the same length")
     end
 
     #Create empty arrays as necessary
     res = Array(Dict{Any, Any}, n)
     isempty(group)? group = ones(Int, n): group
     isempty(y2)? y2 = zeros(Int, n): y2
+    isempty(x2)? x2 = zeros(Int, n): x2
 
     #Concatenate arrays into Dicts
     for i in 1:n
         res[i] = Dict{Any, Any}()
         res[i]["x"] = x[i]
+        res[i]["x2"] = x2[i]
```

Comment:
Fields `x`,`x2`,`y` and `y2` are needed to draw `rect`. 
Previous `barplot` approach used `band` instead of `x2`, which only work for `ordinal` type but not `linear` type.